### PR TITLE
Add: struct ws_object type

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Re-submitted from #27

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
@manuelmessner Please review and try to build, so I know whether it is my machine or another issue.

**Comment by matthiasbeyer** [comment]
@neithernut Rebased and removed all threading related code. Please review, also you, @manuelmessner .

File: src/objects/object.h [hunk 1]
```diff
@@ -28,6 +28,315 @@
...
+
+/**
+ * Object type
+ *
+ * @todo implement locking and refcounting
+ *
+ * The object type is the root class for all classes
+ */
+struct ws_object {
+    ws_object_id* id;        //!< Object id, identifies the actual type
```
Comment (neithernut) [inline src/objects/object.h]: Please rename this to `type` or something.

File: src/objects/object.h [hunk 2]
```diff
@@ -28,6 +28,315 @@
...
+ * Log the object with the passed context
+ *
+ * @memberof ws_object
+ *
+ * @return true if the logging was successful, else false
+ */
+bool
+ws_object_log(
+    struct ws_object const* const self, //!< The object
+    void* /*!< @todo ws_log_context* */
```
Comment (manuelmessner) [inline src/objects/object.h]: There are the wrong sort of comments.
And there is (maybe) a var name missing

Comment (matthiasbeyer) [inline src/objects/object.h]: You're right about the var name, but the comment is absolutely right here.

**Comment by neithernut** [comment]
ACK.

**Comment by manuelmessner** [comment]
ACK

**Pull request merged**